### PR TITLE
[Fset 1173] Assessment centre pass mark entry - Fix the way we store the passmark

### PR DESCRIPTION
**Pull request by mgilgar**
Main part was implemented already. However we want to change the way we store the passmark and this is what PR is about: we will use scheme id (without spaces) instead of scheme name (with spaces) when we store the schemes in mongo.

There are also changes in the following projects:
-fasttrack-admin-frontend
-fasttrack-admin-acceptance-tests

### CODE REVIEW

File: app/controllers/AssessmentCentrePassMarkSettingsController.scala [hunk 1]
```diff
@@ -22,32 +22,39 @@ import model.PassmarkPersistedObjects.Implicits._
...
-  val passmarkService = AssessmentCentrePassMarkSettingsService
+  val assessmentCentrePassMarkRepository = assessmentCentrePassMarkSettingsRepository
+  val assessmentCentrePassmarkService = AssessmentCentrePassMarkSettingsService
+  val auditService = AuditService
 }
 
 trait AssessmentCentrePassMarkSettingsController extends BaseController {
-  val acpmsRepository: AssessmentCentrePassMarkSettingsRepository
-  val passmarkService: AssessmentCentrePassMarkSettingsService
+  val assessmentCentrePassMarkRepository: AssessmentCentrePassMarkSettingsRepository
```
Comment (kjharkin) [inline app/controllers/AssessmentCentrePassMarkSettingsController.scala]: I know this was there before so not your doing but it seems odd that we have the controller have a dependency on a 'service' and a 'repository' here.

File: app/controllers/AssessmentCentrePassMarkSettingsController.scala [hunk 2]
```diff
@@ -22,32 +22,39 @@ import model.PassmarkPersistedObjects.Implicits._
...
   def getLatestVersion = Action.async { implicit request =>
-    passmarkService.getLatestVersion.map { passmark =>
+    assessmentCentrePassmarkService.getLatestVersion.map { passmark =>
       Ok(Json.toJson(passmark))
     }
   }
 
   def create = Action.async(parse.json) { implicit request =>
     withJsonBody[AssessmentCentrePassMarkSettings] { settings =>
-      acpmsRepository.create(settings).map { _ =>
```
Comment (kjharkin) [inline app/controllers/AssessmentCentrePassMarkSettingsController.scala]: Ideally should be moved the service and the dependency from on the repository removed.

File: app/model/OnlineTestCommands.scala [hunk 3]
```diff
@@ -34,10 +34,10 @@ object OnlineTestCommands {
   case class OnlineTestReport(xml: Option[String])
 
   case class CandidateEvaluationData(
-    passmarkSettings: Settings,
-    schemes: List[Scheme],
-    scores: CandidateTestReport,
-    applicationStatus: ApplicationStatuses.EnumVal
+                                      passmarkSettings: OnlineTestPassmarkSettings,
```
Comment (kjharkin) [inline app/model/OnlineTestCommands.scala]: (Minor point) Some funky formatting going on here

File: test/services/evaluation/AssessmentCentreAllSchemesEvaluatorSpec.scala [hunk 4]
```diff
@@ -17,88 +17,93 @@
...
-        Finance -> Red, // alternative scheme
-        ProjectDelivery -> Red // alternative scheme
-      )
-      val alternativeSchemes: List[String] = DigitalAndTechnology :: Finance :: ProjectDelivery :: Nil
-
-      val evaluation = evaluator.evaluateAlternativeSchemes(allSchemesEvaluation, alternativeSchemes)
-
-      evaluation mustBe Amber
-    }
+    // TODO IAN: Please fix the tests once you tackle assessment centre evaluations
```
Comment (kjharkin) [inline test/services/evaluation/AssessmentCentreAllSchemesEvaluatorSpec.scala]: let's not write comments like this.

File: test/services/evaluation/FinalResultEvaluatorSpec.scala [hunk 5]
```diff
@@ -18,119 +18,123 @@ package services.evaluation
...
-  private def assertSchemeEvaluation(expected: List[PerSchemeEvaluation], actual: List[PerSchemeEvaluation]) = {
-    actual.size mustBe expected.size
-    for (i <- expected.indices) {
-      actual(i) mustBe expected(i)
-    }
-  }
-
-  private def mergeResults(onlineTestEvaluation: OnlineTestPassmarkEvaluation, assessmentCentreEvaluation: AssessmentRuleCategoryResult) =
-    evaluator.mergeResults(onlineTestEvaluation, assessmentCentreEvaluation, AllSchemePreferences)
+  // TODO IAN: Please fix the tests once you tackle assessment centre evaluation
```
Comment (kjharkin) [inline test/services/evaluation/FinalResultEvaluatorSpec.scala]: ditto

**Comment by kjharkin** [comment]
There are a few minor things in here which won't stop the merge happening. Good work.

File: app/controllers/AssessmentCentrePassMarkSettingsController.scala (hunk 1 above)
Comment (mgilgar) [inline app/controllers/AssessmentCentrePassMarkSettingsController.scala]: Yes, I totally agree, I will change it when I bump into it next time.

**Pull request merged**